framework/include/restart/DataIO.h

```diff
@@ -28,6 +28,7 @@
 #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
 #  include <type_traits>
 #endif
+#include LIBMESH_INCLUDE_UNORDERED_MAP
```

Comment:
We can go ahead and drop all this libmesh best unordered map stuff. It's only here because unordered maps aren't in C++03 and MooseRandom predates our C++11 requirement. Rather than propogate this old code into new parts of MOOSE, I'd rather just switch MooseRandom to std::unordered_set<> directly.